core/src/test/java/software/amazon/awssdk/util/DateUtilsTest.java

```diff
@@ -15,217 +15,173 @@
         DateUtils.parseIso8601Date(input);
     }
 
     @Test
-    public void test() throws ParseException {
-        Date date = new Date();
-        System.out.println("         formatISO8601Date: " + DateUtils.formatIso8601Date(date));
-        System.out.println("alternateIso8601DateFormat: " + DateUtils.ALTERNATE_ISO_8601_DATE_FORMAT.print(date.getTime()));
-    }
-
-    @Test
     public void testIssue233() throws ParseException {
         // https://github.com/aws/aws-sdk-java/issues/233
-        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-        sdf.setTimeZone(new SimpleTimeZone(0, "GMT"));
-        final String edgeCase = "292278994-08-17T07:12:55.807Z";
-        Date expected = sdf.parse(edgeCase);
+        final String edgeCase = String.valueOf(MAX_MILLIS_YEAR) + "-08-17T07:12:00Z";
+        Instant expected = COMMON_DATE_FORMAT.parse(edgeCase).toInstant();
         if (DEBUG) {
             System.out.println("date: " + expected);
         }
-        String formatted = DateUtils.formatIso8601Date(expected);
+        String formatted = ISO_INSTANT.format(expected);
         if (DEBUG) {
             System.out.println("formatted: " + formatted);
         }
-        assertEquals(edgeCase, formatted);
-        Date parsed = DateUtils.parseIso8601Date(edgeCase);
+        // we have '+' sign as prefix for years. See java.time.format.SignStyle.EXCEEDS_PAD
+        assertEquals(edgeCase, formatted.substring(1));
+
+        Instant parsed = DateUtils.parseIso8601Date(formatted);
         if (DEBUG) {
             System.out.println("parsed: " + parsed);
         }
         assertEquals(expected, parsed);
-        String reformatted = DateUtils.formatIso8601Date(parsed);
-        assertEquals(edgeCase, reformatted);
+        String reformatted = ISO_INSTANT.format(parsed);
+        // we have '+' sign as prefix for years. See java.time.format.SignStyle.EXCEEDS_PAD
+        assertEquals(edgeCase, reformatted.substring(1));
     }
 
-    @Test(expected = IllegalFieldValueException.class)
-    public void testIssue233JodaTimeLimit() throws ParseException {
+    @Test
+    public void testIssue233JavaTimeLimit() {
         // https://github.com/aws/aws-sdk-java/issues/233
-        String s = DateUtils.ISO_8601_DATE_FORMAT.print(Long.MAX_VALUE);
+        String s = ALTERNATE_ISO_8601_DATE_FORMAT.format(
+                ZonedDateTime.ofInstant(Instant.ofEpochMilli(Long.MAX_VALUE), UTC));
         System.out.println("s: " + s);
-        try {
-            DateTime dt = DateUtils.ISO_8601_DATE_FORMAT.parseDateTime(s);
-            fail("Unexpected success: " + dt);
-        } catch (IllegalFieldValueException ex) {
-            // expected
-            throw ex;
-        }
+
+        Instant parsed = DateTimeFormatter.ISO_INSTANT.withZone(UTC).parse(s, Instant::from);
+        assertTrue(ZonedDateTime.ofInstant(parsed, UTC).getYear() == MAX_MILLIS_YEAR);
```

Comment:
It's more correct to use `assertEquals` as the error message will be more helpful if the assertion fails